### DISCUSSION TITLE
Question: How do I access edge attributes?

### DISCUSSION THREAD

**Discussion by Remmargyp** [luukvdmeer/sfnetworks#161] [Q&A] [answered]
Hi! I'm using sfnetworks for a routing algorithm where the inputs are a street-network (shp that is translated to a graph) and a sequence of coordinate-pairs. The algorithm calculates the route between each coordinate pair and creates a dataframe (output) that saves the ids of all edges used by the shortest path. However, I would like to store in the output-dataframe custom ids (called 'LINK_ID', which can be found in the edges of the shapefile). 

If I call the graph (with active edges) by typing its name in the console, I can see that the LINK_ID is stored for each edge, but I don't know how to access it. I know that this is a simple issue but I'm stuck.

Thanks for all input! 

![image](https://user-images.githubusercontent.com/82445576/123107475-df544b80-d439-11eb-9b19-1e5a771caafb.png)

**Comment by loreabad6** [comment, marked as answer]
Hello @Remmargyp thank you for opening your issue. I hope you don't mind that I turned it into a discussion with the Q&A category. It is a bit difficult to help you without a reproducible example (see info on how to create a [reprex here](https://reprex.tidyverse.org/)) and maybe a subset of your data. Can you prepare something like that to illustrate your problem?

**Reply by loreabad6** [reply]
As a first thought of what you are looking for:

```r
graph %>%
    activate("edges") %>%
    as_tibble() %>%
    select(LINK_ID)
```

**Reply by Remmargyp** [reply]
Hey @loreabad6, thank you for your quick reply. I apologize for the lack of a reproducible example, but your reply worked for me!  It gives me a data frame with two columns: LINK_ID and geometry.

To be sure that I'm doing the next step correctly I have the following question: 

> path <- st_network_paths(graph, from = a, to = b)
example_vector <- path$edge_paths

The values that are stored in the example_vector above (a and b being two coordinate pairs); do they correspond to the index of each edge in the graph or to the first attribute stored in the column? >> EDIT: Figured it out: yes they do. 

Thanks a lot!

--- 
A *minor* detail in response to your anwser: I think it's %>%, not %<%.
